invenio_rdm_records/records/api.py

```diff
@@ -333,6 +333,11 @@ def get_files_quota(record=None):
             )
         # Next user quota
         owner_id = record.parent.access.owned_by.owner_id
+        if owner_id is None and hasattr(g, "identity") and hasattr(g.identity, "id"):
+            # Fresh drafts don't have their access properties set yet
+            # (this only happens in the service components)
+            owner_id = g.identity.id
```

Comment:
@slint from our discussions on discord, this is effectively option (2) implemented